config/routes.rb

```diff
@@ -6,12 +6,16 @@
   resources :posts
   resources :comments, only: [:create]
   resources :likes
+  resources :friendships, only: [:create, :destroy, :index]
+  post '/delete_pending_request', to: 'friendships#delete_pending_request'
+  post '/acceptrequest', to: 'friendships#accept_request'
+  post '/deleterequest', to: 'friendships#delete_request'
```

Comment:
**SUGGESTION** 

If you still want to add actions in your controller nonrelated to the default RESTful routes created by rails, this syntaxis might be better:

```
resources :friendships do
  post 'delete_request', on: :member
end
```

or 

```
resources :friendships do
  member do
    post 'delete_request'
    post 'accept_request'
  end
end
```